### PR TITLE
Adapt PR and release jobs to enable testing of the Framework repo

### PR DESCRIPTION
**Pull request by bensze01**
This PR enables the testing of the framework repo, by adapting the existing jobs to allow overriding the version of the framework checked out by the CI, as well as enabling the testing of multiple branches of Mbed TLS in a single job.

The framework version is controlled by the newly defined parameters `FRAMEWORK_REPO` and `FRAMEWORK_BRANCH`, while `MBEDTLS_BRANCH` has been extended to allow specifying multiple branches separated by `,`.

For automatic PR testing, `mbedtls.run_framework_pr_job()` is added as an entrypoint, which is called by the Jenkinsfile added by the companion PR Mbed-TLS/mbedtls-framework#33

Tests:
- Runs with framework version override enabled:
  - [mbed-tls-framework-multibranch][1]
  - [mbed-tls-pr-test-parametrized][2]
  - [mbedtls-release-ci-testing][3]
- Runs without a framework version override:
  - development: [PR-head][4], [PR-merge][5]
  - mbedtls-3.6: [PR-head][6], [PR-merge][7]
  - mbedtls-2.28: [PR-head][8], [PR-merge][9]
  - [mbed-tls-pr-test-parametrized][13]
  - [mbedtls-release-ci-testing][14]
  - Supposed to fail:
    - [ABI Break][10]
    - [Outcome analysis failure][11]
    - [All CMake jobs fail][12]

[1]: https://mbedtls.trustedfirmware.org/job/mbed-tls-framework-multibranch/job/PR-60-head/3/
[2]: https://mbedtls.trustedfirmware.org/job/mbed-tls-restricted-pr-test-parametrized/45/
[3]: https://mbedtls.trustedfirmware.org/job/mbedtls-release-ci-testing/275/
[4]: https://mbedtls.trustedfirmware.org/job/mbed-tls-restricted-pr-ci-testing/job/PR-906-head/25/
[5]: https://mbedtls.trustedfirmware.org/job/mbed-tls-restricted-pr-ci-testing/job/PR-906-merge/26/
[6]: https://mbedtls.trustedfirmware.org/job/mbed-tls-restricted-pr-ci-testing/job/PR-1233-head/5/
[7]: https://mbedtls.trustedfirmware.org/job/mbed-tls-restricted-pr-ci-testing/job/PR-1233-merge/8/
[8]: https://mbedtls.trustedfirmware.org/job/mbed-tls-restricted-pr-ci-testing/job/PR-850-head/14/
[9]: https://mbedtls.trustedfirmware.org/job/mbed-tls-restricted-pr-ci-testing/job/PR-850-merge/12/

[10]: https://mbedtls.trustedfirmware.org/job/mbed-tls-restricted-pr-ci-testing/job/PR-758-merge/12/
[11]: https://mbedtls.trustedfirmware.org/job/mbed-tls-restricted-pr-ci-testing/job/PR-1225-head/4/
[12]: https://mbedtls.trustedfirmware.org/job/mbed-tls-restricted-pr-ci-testing/job/PR-759-head/14/

[13]: https://mbedtls.trustedfirmware.org/job/mbedtls-release-ci-testing/276/
[14]: https://mbedtls.trustedfirmware.org/job/mbed-tls-restricted-pr-test-parametrized/46/

### CODE REVIEW

**Review by minosgalanakis — COMMENTED** [review]
Looks ok overall, with some minor comments.

To my understanding this change will make the ci do two checkous per branch? One for mbedtls and then the framework indipendantly? 

Looking at the combinations of `MBEDTLS_BRANCH` , `FRAMEWORK_BRANCH` times the number of branches, that can cause a significant overhead.

Could we download a non shallow copy of `MBEDTLS_REPO` and share it between jobs, by simply changing the appropriate branch?

File: vars/environ.groovy [hunk 1]
```diff
@@ -45,6 +45,9 @@ void set_pr_environment(String target_repo, boolean is_production) {
 
 def set_common_pr_production_environment() {
     env.CHECKOUT_METHOD = 'scm'
+    /* The credentials here are the SSH credentials for accessing the repositories.
+       They are defined at {JENKINS_URL}/credentials */
+    env.GIT_CREDENTIALS_ID = common.is_open_ci_env ? "mbedtls-github-ssh" : "742b7080-e1cc-41c6-bf55-efb72013bc28"
```
[the remark this answers is not in the document]
Comment (bensze01) [reply vars/environ.groovy:50]: It's just an index, we don't have human-readable key-ids set up on the internal ci at the moment.

**Review by minosgalanakis — DISMISSED** [review]
LGTM

**Review by minosgalanakis — DISMISSED** [review]
LGTM

**Comment by mpg** [comment]
@bensze01 Are the labels needs-ci and needs-work still relevant?

**Comment by bensze01** [comment]
@mpg Needs work is not, needs CI is.

**Comment by bensze01** [comment]
@minosgalanakis @mpg All the CI runs have passed successfully, please (re-)review!

**Review by minosgalanakis — APPROVED** [review]
LGTM

File: vars/analysis.groovy [hunk 2]
```diff
@@ -41,9 +41,6 @@ import net.sf.json.JSONObject
 import org.mbed.tls.jenkins.BranchInfo
 import org.mbed.tls.jenkins.JobTimestamps
 
-// A static field has its content preserved across stages.
```
Comment (mpg) [inline vars/analysis.groovy:44]: Sorry if this should be obvious, but I'm not seeing how the info is preserved accross stages now that it's stored as part of BranchInfo.

**Review by mpg — COMMENTED** [review]
Review in progress; next https://github.com/Mbed-TLS/mbedtls-test/pull/170/commits/dc9d9845b59084cad52c1bd29db832f453db7586

File: vars/analysis.groovy (hunk 2 above)
Comment (mpg) [reply vars/analysis.groovy:44]: (I can see from the test runs that it is preserved, I just can't understand how from looking at the code.)

**Review by mpg — APPROVED** [review]
LGTM, thanks! None of my questions are blockers, so I'm approving this.

**Pull request merged**